services/web-server/src/servers/createApp.js

```diff
@@ -23,7 +26,32 @@ module.exports = async ({ cfg, strategies }) => {
   }).filter(o => o);
   app.use(cors({origin: allowedCORSOrigins}));
 
+  // Use MemoryStore locally (default behavior when a store is not set)
+  const store = process.env.NODE_ENV === 'production' ? AzureTablesStoreFactory.create({
+    table: cfg.azure.tableName,
+    storageAccount: cfg.azure.accountId,
+    accessKey: cfg.azure.accountKey,
+    // Delete expired sessions every 1 hour
+    sessionTimeOut: 60,
+  }) : undefined;
```

Comment:
Bad news incoming:

It looks like `connect-azuretables` doesn't seem to support SAS signatures, which is what we use for controlling access to azure resources.  Providing a storage account and access key means that the application can access any table in that account -- including the secrets service's table, for example.  No bueno..

I also suspect that we will be wanting to add API methods to manage sessions soon (sec folks will want a way to revoke all sessions for a user), and this library's use of a private table wouldn't permit that.

I also suspect that we'll be adding additional Azure tables in web-server, to support the third-party logins process (tables of access_tokens, for example).  So using the existing TC azure infrastructure will make sense.

One option here would be to modify the `connect-azuretables` library to accept a function that returns a SAS signature, which is conveniently what [tc-lib-azure](https://github.com/taskcluster/taskcluster/tree/master/libraries/azure#readme) produces and what [azure-entities expects](https://github.com/taskcluster/azure-entities#sas-function).

Reply:
Adding API methods to manage sessions and additional Azure tables makes a lot of sense.

> One option here would be to modify the connect-azuretables

Would you suggest we fork `connect-azuretables` or create a new library that wraps `connect-azuretables`?

> to accept a function that returns a SAS signature, which is conveniently what tc-lib-azure produces and what azure-entities expects.

I'll look into it.

Reply:
From what I saw, it can't be wrapped (SAS credentials don't last forever so they need to be "fresh", which is why azure-entities supports a callback to get them).  Forking might be a short-term option but we should probably try to upstream the change.  One risk there is is that it seems a bit out of date -- it's using the v2 Azure libraries while v10 is the latest, for example.  It's not clear to me from the docs whether V2 supports SAS signatures.

Reply:
Hmm, perhaps we shouldn't use that library if it's using a very old version of Azure libraries. The list of compatible stores can be found [here](https://github.com/expressjs/session#compatible-session-stores). We could technically create our own Azure store by implementing the required and recommended methods but I fear it might slow us down. Maybe it would make sense to use [memorystore](https://www.npmjs.com/package/memorystore) and accept that we will be logged out when web-server is deployed which shouldn't happen often after this whole login work is done. Thoughts?

Reply:
That seems like a great plan.  Updating or rewriting the Azure connector might be a good project for a contributor during that time, too.